[Build Failure]:

### Version

emacs-plus@30

### Make sure to follow these steps before submitting the issue

- [X] run `brew update` and try to reproduce the issue again
- [X] run `brew doctor`, fix all issues and try to reproduce your issue again
- [X] try to reproduce your issue without any custom options to the formula

### What happened?

<pre>
$ brew install emacs-plus@31 --with-native-comp --with-c9rgreen-sonoma-icon
</pre>

<details>
  <summary>Command output</summary>
  <pre>

==> Fetching d12frosted/emacs-plus/emacs-plus@31
==> Cloning https://github.com/emacs-mirror/emacs.git
Updating /Users/aaronjensen/Library/Caches/Homebrew/emacs-plus@31--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 6ddbe8c7010 * src/w32fns.c (process_dropfiles): Fix cygw32 build (bug#74312).
==> Installing emacs-plus@31 from d12frosted/emacs-plus
==> Patching
==> Applying fix-window-role.patch
==> Applying system-appearance.patch
==> Applying round-undecorated-frame.patch
==> ./autogen.sh
==> ./configure --disable-silent-rules --enable-locallisppath=/opt/homebrew/shar
==> gmake
Last 15 lines from /Users/aaronjensen/Library/Logs/Homebrew/emacs-plus@31/03.gmake:
gmake[1]: *** Waiting for unfinished jobs....
gmake[2]: Leaving directory '/private/var/tmp/emacs-plusA31-20241111-83158-c1cng0/doc/lispref'
gmake[1]: Leaving directory '/private/var/tmp/emacs-plusA31-20241111-83158-c1cng0'
gmake[1]: Entering directory '/private/var/tmp/emacs-plusA31-20241111-83158-c1cng0'
***
*** "make all" failed with exit status 2.
***
*** You could try to:
*** - run "make bootstrap", which might fix the problem
*** - run "make V=1", which displays the full commands invoked by make,
***   to further investigate the problem
***
gmake[1]: *** [Makefile:418: advice-on-failure] Error 2
gmake[1]: Leaving directory '/private/var/tmp/emacs-plusA31-20241111-83158-c1cng0'
gmake: *** [Makefile:374: all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/d12frosted/homebrew-emacs-plus/issues
  </pre>
</details>



<details>
<summary>End of gmake log</summary>
<pre>

Undefined symbols for architecture arm64:
  "_sqlite3_load_extension", referenced from:
      _Fsqlite_load_extension in sqlite.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [Makefile:745: temacs] Error 1
gmake[2]: Leaving directory '/private/var/tmp/emacs-plusA31-20241111-83158-c1cng0/src'
gmake[1]: *** [Makefile:554: src] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake[2]: Leaving directory '/private/var/tmp/emacs-plusA31-20241111-83158-c1cng0/doc/lispref'
gmake[1]: Leaving directory '/private/var/tmp/emacs-plusA31-20241111-83158-c1cng0'
gmake[1]: Entering directory '/private/var/tmp/emacs-plusA31-20241111-83158-c1cng0'
***
*** "make all" failed with exit status 2.
***
*** You could try to:
*** - run "make bootstrap", which might fix the problem
*** - run "make V=1", which displays the full commands invoked by make,
***   to further investigate the problem
***
gmake[1]: *** [Makefile:418: advice-on-failure] Error 2
gmake[1]: Leaving directory '/private/var/tmp/emacs-plusA31-20241111-83158-c1cng0'
gmake: *** [Makefile:374: all] Error 2

HOMEBREW_VERSION: 4.4.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 254bf3fe9d8fa2e1b2fb55dbcf535b2d870180c4
Last commit: 19 hours ago
Core tap JSON: 12 Nov 06:11 UTC
Core cask tap JSON: 12 Nov 06:11 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: ["--appdir=/Applications"]
HOMEBREW_EDITOR: e
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_NO_BOOTSNAP: set
HOMEBREW_TEMP: /private/var/tmp
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 16-core 64-bit dunno
Clang: 16.0.0 build 1600
Git: 2.47.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: N/A
Rosetta 2: false

HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
CFLAGS: -O2 -DFD_SETSIZE=10000 -DDARWIN_UNLIMITED_SELECT -I/opt/homebrew/opt/gcc/include -I/opt/homebrew/opt/libgccjit/include
LDFLAGS: -L/opt/homebrew/lib/gcc/14 -I/opt/homebrew/opt/gcc/include -I/opt/homebrew/opt/libgccjit/include
MAKEFLAGS: -j16
CMAKE_PREFIX_PATH: /opt/homebrew/opt/m4:/opt/homebrew/opt/readline:/opt/homebrew/opt/sqlite:/opt/homebrew/opt/icu4c@76:/opt/homebrew/opt/jpeg:/opt/homebrew/opt/zlib:/opt/homebrew
CMAKE_INCLUDE_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
CMAKE_FRAMEWORK_PATH: /opt/homebrew/opt/python@3.13/Frameworks
PKG_CONFIG_PATH: /opt/homebrew/opt/pkgconf/lib/pkgconfig:/opt/homebrew/opt/xz/lib/pkgconfig:/opt/homebrew/opt/gmp/lib/pkgconfig:/opt/homebrew/opt/libidn2/lib/pkgconfig:/opt/homebrew/opt/libtasn1/lib/pkgconfig:/opt/homebrew/opt/nettle/lib/pkgconfig:/opt/homebrew/opt/p11-kit/lib/pkgconfig:/opt/homebrew/opt/openssl@3/lib/pkgconfig:/opt/homebrew/opt/libevent/lib/pkgconfig:/opt/homebrew/opt/libnghttp2/lib/pkgconfig:/opt/homebrew/opt/unbound/lib/pkgconfig:/opt/homebrew/opt/gnutls/lib/pkgconfig:/opt/homebrew/opt/libpng/lib/pkgconfig:/opt/homebrew/opt/freetype/lib/pkgconfig:/opt/homebrew/opt/fontconfig/lib/pkgconfig:/opt/homebrew/opt/pcre2/lib/pkgconfig:/opt/homebrew/opt/mpdecimal/lib/pkgconfig:/opt/homebrew/opt/readline/lib/pkgconfig:/opt/homebrew/opt/sqlite/lib/pkgconfig:/opt/homebrew/opt/python@3.13/lib/pkgconfig:/opt/homebrew/opt/glib/lib/pkgconfig:/opt/homebrew/opt/libxau/lib/pkgconfig:/opt/homebrew/opt/libxdmcp/lib/pkgconfig:/opt/homebrew/opt/libxcb/lib/pkgconfig:/opt/homebrew/opt/libx11/lib/pkgconfig:/opt/homebrew/opt/libxext/lib/pkgconfig:/opt/homebrew/opt/libxrender/lib/pkgconfig:/opt/homebrew/opt/lzo/lib/pkgconfig:/opt/homebrew/opt/pixman/lib/pkgconfig:/opt/homebrew/opt/cairo/lib/pkgconfig:/opt/homebrew/opt/jpeg-turbo/lib/pkgconfig:/opt/homebrew/opt/lz4/lib/pkgconfig:/opt/homebrew/opt/zstd/lib/pkgconfig:/opt/homebrew/opt/libtiff/lib/pkgconfig:/opt/homebrew/opt/gdk-pixbuf/lib/pkgconfig:/opt/homebrew/opt/fribidi/lib/pkgconfig:/opt/homebrew/opt/graphite2/lib/pkgconfig:/opt/homebrew/opt/icu4c@76/lib/pkgconfig:/opt/homebrew/opt/harfbuzz/lib/pkgconfig:/opt/homebrew/opt/pango/lib/pkgconfig:/opt/homebrew/opt/librsvg/lib/pkgconfig:/opt/homebrew/opt/little-cms2/lib/pkgconfig:/opt/homebrew/opt/jansson/lib/pkgconfig:/opt/homebrew/opt/tree-sitter/lib/pkgconfig:/opt/homebrew/opt/webp/lib/pkgconfig:/opt/homebrew/opt/isl/lib/pkgconfig:/opt/homebrew/opt/mpfr/lib/pkgconfig:/opt/homebrew/opt/jpeg/lib/pkgconfig:/opt/homebrew/opt/zlib/lib/pkgconfig:/opt/homebrew/opt/xorgproto/share/pkgconfig
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/opt/homebrew/Library/Homebrew/os/mac/pkgconfig/15
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_GIT: git
HOMEBREW_SDKROOT: /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk
ACLOCAL_PATH: /opt/homebrew/share/aclocal
PATH: /opt/homebrew/Library/Homebrew/shims/mac/super:/opt/homebrew/opt/make/bin:/opt/homebrew/opt/autoconf/bin:/opt/homebrew/opt/gnu-sed/bin:/opt/homebrew/opt/gnu-tar/bin:/opt/homebrew/opt/grep/bin:/opt/homebrew/opt/awk/bin:/opt/homebrew/opt/coreutils/bin:/opt/homebrew/opt/pkgconf/bin:/opt/homebrew/opt/texinfo/bin:/opt/homebrew/opt/xz/bin:/opt/homebrew/opt/m4/bin:/opt/homebrew/opt/gettext/bin:/opt/homebrew/opt/libidn2/bin:/opt/homebrew/opt/libtasn1/bin:/opt/homebrew/opt/nettle/bin:/opt/homebrew/opt/p11-kit/bin:/opt/homebrew/opt/openssl@3/bin:/opt/homebrew/opt/libevent/bin:/opt/homebrew/opt/gnutls/bin:/opt/homebrew/opt/libpng/bin:/opt/homebrew/opt/freetype/bin:/opt/homebrew/opt/fontconfig/bin:/opt/homebrew/opt/pcre2/bin:/opt/homebrew/opt/sqlite/bin:/opt/homebrew/opt/python@3.13/bin:/opt/homebrew/opt/glib/bin:/opt/homebrew/opt/cairo/bin:/opt/homebrew/opt/jpeg-turbo/bin:/opt/homebrew/opt/lz4/bin:/opt/homebrew/opt/zstd/bin:/opt/homebrew/opt/libtiff/bin:/opt/homebrew/opt/gdk-pixbuf/bin:/opt/homebrew/opt/fribidi/bin:/opt/homebrew/opt/graphite2/bin:/opt/homebrew/opt/icu4c@76/bin:/opt/homebrew/opt/harfbuzz/bin:/opt/homebrew/opt/pango/bin:/opt/homebrew/opt/librsvg/bin:/opt/homebrew/opt/little-cms2/bin:/opt/homebrew/opt/tree-sitter/bin:/opt/homebrew/opt/giflib/bin:/opt/homebrew/opt/webp/bin:/opt/homebrew/opt/gcc/bin:/opt/homebrew/opt/jpeg/bin:/opt/homebrew/opt/python@3.13/libexec/bin:/usr/bin:/bin:/usr/sbin:/sbin
</pre>
</details>


### brew doctor output

Your system is ready to brew.

### brew config output

HOMEBREW_VERSION: 4.4.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 254bf3fe9d8fa2e1b2fb55dbcf535b2d870180c4
Last commit: 19 hours ago
Core tap JSON: 12 Nov 06:11 UTC
Core cask tap JSON: 12 Nov 06:11 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: ["--appdir=/Applications"]
HOMEBREW_EDITOR: e
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_TEMP: /var/tmp
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 16-core 64-bit dunno
Clang: 16.0.0 build 1600
Git: 2.47.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: N/A
Rosetta 2: false

### Any extra information

Brand new laptop.

```
Undefined symbols for architecture arm64:
"_sqlite3_load_extension", referenced from:
_Fsqlite_load_extension in sqlite.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Ooof, initially I thought that it could be an upstream issue. But then I though - do you have sqlite3 installed? We don't have it as a dependency, and this is a fresh install. So maybe that's the reason. For sqlite extensions you need at least 3.13 version (afaik).

--- a/Formula/emacs-plus@29.rb
+++ b/Formula/emacs-plus@29.rb
@@ -44,6 +44,7 @@ class EmacsPlusAT29 < EmacsBase
   depends_on "texinfo" => :build
   depends_on "xz" => :build
   depends_on "m4" => :build
+  depends_on "sqlite" => :build
   depends_on "gnutls"
   depends_on "librsvg"
   depends_on "little-cms2"
@@ -113,6 +114,9 @@ class EmacsPlusAT29 < EmacsBase
 
     ENV.append "CFLAGS", "-g -Og" if build.with? "debug"
     ENV.append "CFLAGS", "-O2 -DFD_SETSIZE=10000 -DDARWIN_UNLIMITED_SELECT"
+
+    ENV.append "CFLAGS", "-I#{Formula["sqlite"].include}"
+    ENV.append "LDFLAGS", "-L#{Formula["sqlite"].opt_lib}"
 
     # Necessary for libgccjit library discovery
     if build.with? "native-comp"

--- a/Formula/emacs-plus@30.rb
+++ b/Formula/emacs-plus@30.rb
@@ -48,6 +48,7 @@ class EmacsPlusAT30 < EmacsBase
   depends_on "texinfo" => :build
   depends_on "xz" => :build
   depends_on "m4" => :build
+  depends_on "sqlite" => :build
   depends_on "gnutls"
   depends_on "librsvg"
   depends_on "little-cms2"
@@ -126,6 +127,9 @@ class EmacsPlusAT30 < EmacsBase
 
     ENV.append "CFLAGS", "-g -Og" if build.with? "debug"
     ENV.append "CFLAGS", "-O2 -DFD_SETSIZE=10000 -DDARWIN_UNLIMITED_SELECT"
+
+    ENV.append "CFLAGS", "-I#{Formula["sqlite"].include}"
+    ENV.append "LDFLAGS", "-L#{Formula["sqlite"].opt_lib}"
 
     # Necessary for libgccjit library discovery
     if build.with? "native-comp"

--- a/Formula/emacs-plus@31.rb
+++ b/Formula/emacs-plus@31.rb
@@ -37,7 +37,8 @@ class EmacsPlusAT31 < EmacsBase
   depends_on "pkg-config" => :build
   depends_on "texinfo" => :build
   depends_on "xz" => :build
-  depends_on "m4" => :build 
+  depends_on "m4" => :build
+  depends_on "sqlite" => :build
   depends_on "gnutls"
   depends_on "librsvg"
   depends_on "little-cms2"
@@ -118,6 +119,9 @@ class EmacsPlusAT31 < EmacsBase
 
     ENV.append "CFLAGS", "-g -Og" if build.with? "debug"
     ENV.append "CFLAGS", "-O2 -DFD_SETSIZE=10000 -DDARWIN_UNLIMITED_SELECT"
+
+    ENV.append "CFLAGS", "-I#{Formula["sqlite"].include}"
+    ENV.append "LDFLAGS", "-L#{Formula["sqlite"].opt_lib}"
 
     # Necessary for libgccjit library discovery
     if build.with? "native-comp"

